File: net/pdfjet/Token.cs
using System;
using System.Text;

namespace PDFjet.NET {
/// <summary>Byte sequences of the PDF syntax.</summary>
public class Token {
    // Fundamental structural tokens
    /// <summary>A space.</summary>
    public static readonly byte Space = (byte) ' ';
    /// <summary>A line feed.</summary>
    public static readonly byte Newline = (byte) '\n';
    /// <summary>The start of a dictionary: &lt;&lt;</summary>
    public static readonly byte[] BeginDictionary = Encoding.ASCII.GetBytes("<<\n");
    /// <summary>The end of a dictionary: &gt;&gt;</summary>
    public static readonly byte[] EndDictionary = Encoding.ASCII.GetBytes(">>\n");
    /// <summary>The stream keyword.</summary>
    public static readonly byte[] Stream = Encoding.ASCII.GetBytes("stream\n");
    /// <summary>The endstream keyword.</summary>
    public static readonly byte[] EndStream = Encoding.ASCII.GetBytes("\nendstream\n");

    // Object management tokens
    /// <summary>The " 0 obj" that follows an object number.</summary>
    public static readonly byte[] NewObj = Encoding.ASCII.GetBytes(" 0 obj\n");
    /// <summary>The endobj keyword.</summary>
    public static readonly byte[] EndObj = Encoding.ASCII.GetBytes("endobj\n");
    /// <summary>The " 0 R" that follows the number of a referenced object.</summary>
    public static readonly byte[] ObjRef = Encoding.ASCII.GetBytes(" 0 R\n");

    // Text and content tokens
    /// <summary>The BT operator, which begins a text object.</summary>
    public static readonly byte[] BeginText = Encoding.ASCII.GetBytes("BT\n");
    /// <summary>The ET operator, which ends a text object.</summary>
    public static readonly byte[] EndText = Encoding.ASCII.GetBytes("ET\n");

    // Essential property tokens (used everywhere)
    /// <summary>The /Length key.</summary>
    public static readonly byte[] Length = Encoding.ASCII.GetBytes("/Length ");
    /// <summary>The /Type key.</summary>
    public static readonly byte[] Type = Encoding.ASCII.GetBytes("/Type ");
    /// <summary>The /Resources key.</summary>
    public static readonly byte[] Resources = Encoding.ASCII.GetBytes("/Resources ");
}
}

File: Sources/PDFjet/Token.swift
/**
 * Token.swift
 *
 * Copyright (c) 2026 PDFjet Software
 * Licensed under the MIT License. See LICENSE file in the project root.
 */

///
/// Byte sequences of the PDF syntax, internal as in the other ports.
/// Please see PDF.swift
///
class Token {
    // Fundamental structural tokens
    /// A space.
    static let space: UInt8 = 32     // ASCII space
    /// A line feed.
    static let newline: UInt8 = 10   // ASCII LF

    /// The start of a dictionary: <<
    static let beginDictionary = [UInt8]("<<\n".utf8)
    /// The end of a dictionary: >>
    static let endDictionary = [UInt8](">>\n".utf8)
    /// The stream keyword.
    static let stream = [UInt8]("stream\n".utf8)
    /// The endstream keyword.
    static let endStream = [UInt8]("\nendstream\n".utf8)

    // Object management tokens
    /// The " 0 obj" that follows an object number.
    static let newObj = [UInt8](" 0 obj\n".utf8)
    /// The endobj keyword.
    static let endObj = [UInt8]("endobj\n".utf8)
    /// The " 0 R" that follows the number of a referenced object.
    static let objRef = [UInt8](" 0 R\n".utf8)

    // Text and content tokens
    /// The BT operator, which begins a text object.
    static let beginText = [UInt8]("BT\n".utf8)
    /// The ET operator, which ends a text object.
    static let endText = [UInt8]("ET\n".utf8)

    // Essential property tokens (used everywhere)
    /// The /Length key.
    static let length = [UInt8]("/Length ".utf8)
    /// The /Type key.
    static let type = [UInt8]("/Type ".utf8)
    /// The /Resources key.
    static let resources = [UInt8]("/Resources ".utf8)
}
